packages/openapi-fetch/test/middleware/middleware.test.ts

```diff
@@ -505,3 +505,118 @@ test("skips onResponse handlers when response is returned from onRequest", async
 
   expect(onResponseCalled).toBe(false);
 });
+
+test('it should enable a middleware to be added via the "middleware" request option', async () => {
+  let actualRequest = new Request("https://nottherealurl.fake");
+  const client = createObservedClient<paths>({}, async (req) => {
+    actualRequest = new Request(req);
+    return Response.json({});
+  });
+
+  await client.GET("/posts/{id}", {
+    params: { path: { id: 123 } },
+    middleware: [
+      {
+        async onRequest({ request }) {
+          return new Request("https://foo.bar/api/v1", {
+            ...request,
+            method: "OPTIONS",
+            headers: { foo: "bar" },
+          });
+        },
+      },
+    ],
+  });
+
+  expect(actualRequest.url).toBe("https://foo.bar/api/v1");
+  expect(actualRequest.method).toBe("OPTIONS");
+  expect(actualRequest.headers.get("foo")).toBe("bar");
```

Comment:
Actually test reference equality of the response, just like here?

https://github.com/Kauabunga/openapi-typescript/blob/7527d1e7502cd1e9621922f028b4736d85f25800/packages/openapi-fetch/test/middleware/middleware.test.ts#L447-L465